langchain4j-qdrant/src/main/java/dev/langchain4j/store/embedding/qdrant/QdrantEmbeddingStore.java

```diff
@@ -0,0 +1,310 @@
+package dev.langchain4j.store.embedding.qdrant;
+
+import static dev.langchain4j.internal.Utils.randomUUID;
+import static io.qdrant.client.PointIdFactory.id;
+import static io.qdrant.client.ValueFactory.value;
+import static io.qdrant.client.VectorsFactory.vectors;
+import static io.qdrant.client.WithPayloadSelectorFactory.enable;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.singletonList;
+import static java.util.Comparator.comparingDouble;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toMap;
+
+import dev.langchain4j.data.document.Metadata;
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.store.embedding.CosineSimilarity;
+import dev.langchain4j.store.embedding.EmbeddingMatch;
+import dev.langchain4j.store.embedding.EmbeddingStore;
+import dev.langchain4j.store.embedding.RelevanceScore;
+import io.qdrant.client.QdrantClient;
+import io.qdrant.client.QdrantGrpcClient;
+import io.qdrant.client.WithVectorsSelectorFactory;
+import io.qdrant.client.grpc.JsonWithInt.Value;
+import io.qdrant.client.grpc.Points.DeletePoints;
+import io.qdrant.client.grpc.Points.Filter;
+import io.qdrant.client.grpc.Points.PointStruct;
+import io.qdrant.client.grpc.Points.PointsSelector;
+import io.qdrant.client.grpc.Points.ScoredPoint;
+import io.qdrant.client.grpc.Points.SearchPoints;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.concurrent.ExecutionException;
+import javax.annotation.Nullable;
+
+/**
+ * Represents a <a href="https://qdrant.tech/">Qdrant</a> collection as an embedding store. With
+ * support for storing {@link dev.langchain4j.data.document.Metadata}.
+ */
+public class QdrantEmbeddingStore implements EmbeddingStore<TextSegment> {
+
+  private final QdrantClient client;
+  private final String payloadTextKey;
+  private final String collectionName;
+
+  /**
+   * @param collectionName The name of the Qdrant collection.
+   * @param host The host of the Qdrant instance.
+   * @param port The GRPC port of the Qdrant instance.
+   * @param useTls Whether to use TLS(HTTPS).
+   * @param payloadTextKey The field name of the text segment in the Qdrant payload.
+   * @param apiKey The Qdrant API key to authenticate with.
+   */
+  public QdrantEmbeddingStore(
```

Comment:
I would add another ctor (and builder method) that accepts `QdrantClient`, to give user more flexibility

Reply:
https://github.com/langchain4j/langchain4j/pull/542/commits/c050861ec070430c4b35e30f1e107af0192e9a4c

Reply:
https://github.com/langchain4j/langchain4j/pull/542/commits/e068beb5362ea90fedef09bb9aa9df524fd07965